nat/mapping/port_mapping.go

```diff
@@ -81,16 +86,17 @@ func mapPort(m portmap.Interface, c chan struct{}, protocol string, extPort, int
 	}
 }
 
-func addMapping(m portmap.Interface, protocol string, extPort, intPort int, name string, publisher Publisher) {
+func addMapping(m portmap.Interface, protocol string, extPort, intPort int, name string, publisher Publisher) error {
 	if err := m.AddMapping(protocol, extPort, intPort, name, mapTimeout); err != nil {
 		log.Debugf("%s, Couldn't add port mapping for port %d: %v, retrying with permanent lease", logPrefix, extPort, err)
 		if err := m.AddMapping(protocol, extPort, intPort, name, 0); err != nil {
 			// some gateways support only permanent leases
-			publisher.Publish(traversal.EventTopic, traversal.EventFailure)
+			publisher.Publish(traversal.EventTopic, traversal.BuildFailureEvent(err))
 			log.Debugf("%s Couldn't add port mapping for port %d: %v", logPrefix, extPort, err)
-			return
+			return err
 		}
 	}
-	publisher.Publish(traversal.EventTopic, traversal.EventSuccess)
+	publisher.Publish(traversal.EventTopic, traversal.BuildSuccessEvent())
```

Comment:
We could discriminate between what kind of success: with mapTimeout or with persistant (0). This would give us additional information in the future.

Reply:
I think we've agreed to add more stuff later, when/if needed.